### PR TITLE
rework compare.py and use hashlib for hashing

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
```
$ ~/dev/cbt/compare.py -a /tmp/ -b /tmp/master -v
18:36:20 - DEBUG    - cbt      - Nodes : gen8
18:36:20 - DEBUG    - cbt      - Nodes : gen8
18:36:20 - DEBUG    - cbt      - Nodes : gen8
18:36:20 - WARNING  - cbt      - prefill/gen8/0: bandwidth: (or (greater) (near 0.05)):: 0.176259/0.186155  => rejected
18:36:20 - INFO     - cbt      - prefill/gen8/0: iops_avg: (or (greater) (near 0.05)):: 45.0/47.0  => accepted
18:36:20 - WARNING  - cbt      - prefill/gen8/0: iops_stddev: (or (less) (near 0.05)):: 8.32666/6.65833  => rejected
18:36:20 - WARNING  - cbt      - prefill/gen8/0: latency_avg: (or (less) (near 0.05)):: 0.353641/0.333712  => rejected
18:36:20 - INFO     - cbt      - prefill/gen8/1: bandwidth: (or (greater) (near 0.05)):: 0.186733/0.177619  => accepted
18:36:20 - INFO     - cbt      - prefill/gen8/1: iops_avg: (or (greater) (near 0.05)):: 47.0/45.0  => accepted
18:36:20 - INFO     - cbt      - prefill/gen8/1: iops_stddev: (or (less) (near 0.05)):: 6.65833/9.81495  => accepted
18:36:20 - INFO     - cbt      - prefill/gen8/1: latency_avg: (or (less) (near 0.05)):: 0.332307/0.350251  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/0: bandwidth: (or (greater) (near 0.05)):: 1.20049/1.22336  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/0: iops_avg: (or (greater) (near 0.05)):: 307.0/313.0  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/0: iops_stddev: (or (less) (near 0.05)):: 0.0/0.0  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/0: latency_avg: (or (less) (near 0.05)):: 0.0519279/0.0509029  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/1: bandwidth: (or (greater) (near 0.05)):: 1.31387/1.11372  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/1: iops_avg: (or (greater) (near 0.05)):: 336.0/285.0  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/1: iops_stddev: (or (less) (near 0.05)):: 0.0/0.0  => accepted
18:36:20 - INFO     - cbt      - seq/gen8/1: latency_avg: (or (less) (near 0.05)):: 0.0469234/0.0557337  => accepted
18:36:20 - WARNING  - cbt      - 3 tests failed out of 16
```

a sample output.

File: benchmark/benchmark.py [hunk 1]
```diff
@@ -15,10 +16,12 @@ def __init__(self, archive_dir, cluster, config):
         self.acceptable = config.pop('acceptable', [])
         self.config = config
         self.cluster = cluster
+        hashable = json.dumps(sorted(self.config.items())).encode()
+        digest = hashlib.sha1(hashable).hexdigest()[:8]
         self.archive_dir = os.path.join(archive_dir,
                                         'results',
                                         '{:0>8}'.format(config.get('iteration')),
-                                        'id{}'.format(hash(frozenset((self.config).items()))))
+                                        'id-{}'.format(digest))
```
Comment (rzarzynski) [inline benchmark/benchmark.py]: This change isn't backward compatible but I doubt it's an issue.

File: benchmark/benchmark.py [hunk 2]
```diff
@@ -98,3 +98,18 @@ def dropcaches(self):
...
+        self.result = result
+        self.baseline = baseline
+        self.stmt = stmt
+        self.accepted = accepted
+
+    def __str__(self):
+        fmt = '{run}: {alias}: {stmt}:: {result}/{baseline}  => {accepted}'
+        return fmt.format(run=self.run, alias=self.alias, stmt=self.stmt,
+                          result=self.result, baseline=self.baseline,
+                          accepted="accepted" if self.accepted else "rejected")
```
Comment (rzarzynski) [inline benchmark/benchmark.py]: nit: rename the `accepted` format parameter to e.g. `status`?

Comment (tchaikov) [reply benchmark/benchmark.py]: fixed and repushed.

**Pull request merged**